### ISSUE TITLE
MX Master 3 rules not working on fresh install of ubuntu 24

### ISSUE THREAD

**Issue by jakebottrall** [pwr-Solaar/Solaar#2648]
**Information**

- Solaar version (`solaar --version` or `git describe --tags` if cloned from this repository): solaar 1.1.13
- Distribution: Ubuntu 24.04.1 LTS
- Kernel version (ex. `uname -srmo`): `Linux 6.8.0-47-generic x86_64 GNU/Linux`
- Output of `solaar show`:

<details>

```
rules cannot access modifier keys in Wayland, accessing process only works on GNOME with Solaar Gnome extension installed
solaar version 1.1.13

Unifying Receiver
  Device path  : /dev/hidraw3
  USB id       : 046d:C52B
  Serial       : 347F9111
  C Pending    : ff
    Firmware   : 12.11.B0032
    Bootloader : 04.16
    Other      : AA.AA
  Has 1 paired device(s) out of a maximum of 6.
  Notifications: wireless, software present (0x000900)
  Device activity counters: 1=85

  1: MX Master 3 Wireless Mouse
     Device path  : /dev/hidraw2
     WPID         : 4082
     Codename     : MX Master 3
     Kind         : mouse
     Protocol     : HID++ 4.5
     Report Rate : 8ms
     Serial number: 7605D99F
     Model ID:      B02340820000
     Unit ID:       7605D99F
                 1: BOT 95.00.B0013
                 0: MPM 19.01.B0015
                 3: 
     The power switch is located on the base.
     Supports 35 HID++ 2.0 features:
         0: ROOT                   {0000} V0     
         1: FEATURE SET            {0001} V0     
         2: DEVICE FW VERSION      {0003} V3     
            Firmware: Bootloader BOT 95.00.B0013 0000D8279515
            Firmware: Firmware MPM 19.01.B0015 40826A9CA89D
            Firmware: Other   
            Unit ID: 7605D99F  Model ID: B02340820000  Transport IDs: {'btleid': 'B023', 'wpid': '4082'}
         3: DEVICE NAME            {0005} V0     
            Name: Wireless Mouse MX Master 3
            Kind: mouse
         4: WIRELESS DEVICE STATUS {1D4B} V0     
         5: CONFIG CHANGE          {0020} V0     
            Configuration: 11000000000000000000000000000000
         6: CRYPTO ID              {0021} V1     
         7: DEVICE FRIENDLY NAME   {0007} V0     
            Friendly Name: MX Master 3
         8: BATTERY STATUS         {1000} V1     
            Battery: 50%, discharging, next level 20%.
         9: REPROG CONTROLS V4     {1B04} V4     
            Key/Button Actions (saved): {Left Button:Left Click, Right Button:Right Click, Middle Button:Mouse Middle Button, Back Button:Mouse Back Button, Forward Button:Mouse Forward Button, Mouse Gesture Button:Gesture Button Navigation, Smart Shift:Smart Shift}
            Key/Button Actions        : {Left Button:Left Click, Right Button:Right Click, Middle Button:Mouse Middle Button, Back Button:Mouse Back Button, Forward Button:Mouse Forward Button, Mouse Gesture Button:Gesture Button Navigation, Smart Shift:Smart Shift}
            Key/Button Diversion (saved): {Middle Button:Regular, Back Button:Regular, Forward Button:Regular, Mouse Gesture Button:Regular, Smart Shift:Regular}
            Key/Button Diversion        : {Middle Button:Regular, Back Button:Regular, Forward Button:Regular, Mouse Gesture Button:Regular, Smart Shift:Regular}
        10: CHANGE HOST            {1814} V1     
            Change Host        : 1:poppy-puter
        11: XY STATS               {2250} V1     
        12: ADJUSTABLE DPI         {2201} V1     
            Sensitivity (DPI) (saved): 1000
            Sensitivity (DPI)        : 1000
        13: SMART SHIFT            {2110} V0     
            Scroll Wheel Ratcheted (saved): Ratcheted
            Scroll Wheel Ratcheted        : Ratcheted
            Scroll Wheel Ratchet Speed (saved): 10
            Scroll Wheel Ratchet Speed        : 10
        14: HIRES WHEEL            {2121} V1     
            Multiplier: 15
            Has invert: Normal wheel motion
            Has ratchet switch: Normal wheel mode
            High resolution mode
            HID notification
            Scroll Wheel Direction (saved): False
            Scroll Wheel Direction        : False
            Scroll Wheel Resolution (saved): True
            Scroll Wheel Resolution        : True
            Scroll Wheel Diversion (saved): False
            Scroll Wheel Diversion        : False
        15: THUMB WHEEL            {2150} V0     
            Thumb Wheel Direction (saved): False
            Thumb Wheel Direction        : False
            Thumb Wheel Diversion (saved): False
            Thumb Wheel Diversion        : False
        16: WHEEL STATS            {2251} V0     
        17: DFUCONTROL SIGNED      {00C2} V0     
        18: DEVICE RESET           {1802} V0    internal, hidden 
        19: unknown:1803           {1803} V0    internal, hidden 
        20: CONFIG DEVICE PROPS    {1806} V6    internal, hidden 
        21: unknown:1813           {1813} V0    internal, hidden 
        22: OOBSTATE               {1805} V0    internal, hidden 
        23: unknown:1830           {1830} V0    internal, hidden 
        24: unknown:1890           {1890} V5    internal, hidden 
        25: unknown:1891           {1891} V5    internal, hidden 
        26: unknown:18A1           {18A1} V0    internal, hidden 
        27: unknown:1DF3           {1DF3} V0    internal, hidden 
        28: unknown:1E00           {1E00} V0    hidden 
        29: unknown:1EB0           {1EB0} V0    internal, hidden 
        30: unknown:1861           {1861} V0    internal, hidden 
        31: unknown:9001           {9001} V0    internal, hidden 
        32: unknown:9203           {9203} V0    internal, hidden 
        33: unknown:9205           {9205} V0    internal, hidden 
        34: unknown:9300           {9300} V0    internal, hidden 
     Has 8 reprogrammable keys:
         0: Left Button               , default: Left Click                  => Left Click                
             mse, analytics key events, pos:0, group:1, group mask:g1
             reporting: default
         1: Right Button              , default: Right Click                 => Right Click               
             mse, analytics key events, pos:0, group:1, group mask:g1
             reporting: default
         2: Middle Button             , default: Mouse Middle Button         => Mouse Middle Button       
             mse, reprogrammable, divertable, raw XY, analytics key events, pos:0, group:3, group mask:g1,g2,g3
             reporting: default
         3: Back Button               , default: Mouse Back Button           => Mouse Back Button         
             mse, reprogrammable, divertable, raw XY, analytics key events, pos:0, group:2, group mask:g1,g2
             reporting: default
         4: Forward Button            , default: Mouse Forward Button        => Mouse Forward Button      
             mse, reprogrammable, divertable, raw XY, analytics key events, pos:0, group:2, group mask:g1,g2
             reporting: default
         5: Mouse Gesture Button      , default: Gesture Button Navigation   => Gesture Button Navigation 
             mse, reprogrammable, divertable, raw XY, analytics key events, pos:0, group:3, group mask:g1,g2,g3
             reporting: default
         6: Smart Shift               , default: Smart Shift                 => Smart Shift               
             mse, reprogrammable, divertable, raw XY, analytics key events, pos:0, group:3, group mask:g1,g2,g3
             reporting: default
         7: Virtual Gesture Button    , default: Virtual Gesture Button      => Virtual Gesture Button    
             divertable, virtual, raw XY, force raw XY, pos:0, group:4, group mask:empty
             reporting: default
     Battery: 50%, discharging, next level 20%.
```
</details>

- Contents of `~/.config/solaar/config.yaml` (or `~/.config/solaar/config.json` if `~/.config/solaar/config.yaml` not present):

<details>

```
- 1.1.13
- _NAME: MX Master 3 Wireless Mouse
  _absent: [hi-res-scroll, lowres-scroll-mode, onboard_profiles, report_rate, report_rate_extended, pointer_speed, dpi_extended, speed-change, backlight,
    backlight_level, backlight_duration_hands_out, backlight_duration_hands_in, backlight_duration_powered, backlight-timed, led_control, led_zone_, rgb_control,
    rgb_zone_, brightness_control, per-key-lighting, fn-swap, persistent-remappable-keys, disable-keyboard-keys, crown-smooth, divert-crown, divert-gkeys,
    m-key-leds, mr-key-led, multiplatform, gesture2-gestures, gesture2-divert, gesture2-params, sidetone, equalizer, adc_power_management]
  _battery: 4096
  _modelId: B02340820000
  _sensitive: {change-host: false, hires-scroll-mode: false, hires-smooth-invert: false, hires-smooth-resolution: false}
  _serial: 7605D99F
  _unitId: 7605D99F
  _wpid: '4082'
  change-host: null
  divert-keys: {82: 0, 83: 0, 86: 0, 195: 0, 196: 0}
  dpi: 1000
  hires-scroll-mode: false
  hires-smooth-invert: false
  hires-smooth-resolution: true
  reprogrammable-keys: {80: 80, 81: 81, 82: 82, 83: 83, 86: 86, 195: 195, 196: 196}
  scroll-ratchet: 2
  smart-shift: 10
  thumb-scroll-invert: false
  thumb-scroll-mode: false
```
</details>

- Errors or warrnings from Solaar:

No errors or warnings in `~/.tmp` or `solaar -dd`.

**Describe the bug**
I'm unable to create rules for the gesture buttons for my MX Master 3. I've followed the instructions set out in https://pwr-solaar.github.io/Solaar/rules/ with no luck. I do notice that I'm getting the following message:
```
rules cannot access modifier keys in Wayland, accessing process only works on GNOME with Solaar Gnome extension installed
```
However I do have the gnome extension installed and enabled. So I'm not sure if this means it's not connecting to the extension properly or if this is a standard message when it detects your using wayland.

No matter what I do the gesture button will always map to "super"+"tab". I've confirmed this by changing this in my keyboard shortcuts.

Also note this is on a fresh install of Ubuntu 24. Everything worked for before reinstalling. I was also on Ubuntu 24 beforehand. Here are my rules (copied over from my previous install):
```
%YAML 1.3
---
- MouseGesture: Mouse Gesture Button
- KeyPress:
  - Super_L
  - click
...
---
- MouseGesture: Mouse Right
- KeyPress:
  - [Super_R, Page_Up]
  - click
...
---
- MouseGesture: Mouse Left
- KeyPress:
  - [Super_R, Page_Down]
  - click
...
```

**To Reproduce**
1. Start a fresh install of Ubuntu 24
2. Install Solaar
3. Follow instructions in https://pwr-solaar.github.io/Solaar/rules/
4. Add rules for gesture button
5. Gesture button always defaults to "super"+"tab"

**Screenshots**
n/a

**Additional context**
n/a

**Comment by pfps** [comment]
You have to divert a button to mouse gestures before mouse gestures will work.

**Comment by jakebottrall** [comment]
@pfps oh man I can't believe I missed that, this has been driving me mental all day long. Thankyou for the assistance!

**Comment by glennDittmann** [comment]
Just a follow up question: 
I used the GUI version of Solaar to make rules for my MX 3S gesture buttons.

Then following your discussion here I had to set the Key/Button Diversion so the rules actually get triggered/activated, s. Screenshot.

![Image](https://github.com/user-attachments/assets/71069995-1dba-4a59-a44d-88c72a4e0859)

Just out of curiosity: what does this acutally mean technically. I see the tooltip say: "_Make the key or button send HID++ notifications (Diverted) or initiate Mouse Gestures or Sliding DPI_"

Does it mean in case of the `Mouse Gesture Button` it is actually just an `enabled`-flag or does something get diverted as well..?
